google/cloud/datastore/client.py

```diff
@@ -199,8 +199,22 @@ def _extended_lookup(
 
     results = []
 
+    transaction_id = None
+    new_transaction_options = None
+    if transaction is not None:
+        transaction_id = transaction.id
+        if (
+            transaction_id is None
+            and transaction._begin_later
+            and transaction._status == transaction._INITIAL
+        ):
+            # if transaction hasn't been initialized, initialize it as part of this request
+            new_transaction_options = transaction._options
+
     loop_num = 0
-    read_options = helpers.get_read_options(eventual, transaction_id, read_time)
+    read_options = helpers.get_read_options(
+        eventual, transaction_id, read_time, new_transaction_options
+    )
     while loop_num < _MAX_LOOPS:  # loop against possible deferred.
         loop_num += 1
         request = {
```

Comment:
How does this ensure that two requests are not being made at the same time? Or is the strategy here to error out with `Transaction already begun` on the second response instead of using a mutex to block one request while the other transaction request is in progress?